order/rpc_parse.go

```diff
@@ -115,6 +115,22 @@ func ParseRPCOrder(version, leaseDuration uint32,
 			details.ChannelType)
 	}
 
+	if len(details.AllowedNodeIds) > 0 &&
+		len(details.NotAllowedNodeIds) > 0 {
+
+		return nil, errors.New("my story")
+	}
+
+	kit.AllowedNodeIDs = make([][33]byte, len(details.AllowedNodeIds))
+	for idx, nodeID := range details.AllowedNodeIds {
```

Comment:
We should parse the public keys here to make sure no invalid data is given to the RPC call.